packages/core/bootstrap/test/unit/logger.test.ts

```diff
@@ -17,5 +23,27 @@ describe('Logger', () => {
         'http://test.com/endpoint?client=REDACTED',
       )
     })
+    it('properly builds censor list', () => {
+      const censorList = CensorList.getAll()
+      expect(censorList[0]).toEqual({ key: 'API_KEY', value: RegExp('mock\\-api\\-key', 'gi') })
+    })
+    it('properly redacts API_KEY (string)', () => {
+      const redacted = censorLog('mock-api-key', CensorList.getAll())
+      expect(redacted).toEqual('[API_KEY REDACTED]')
+    })
+    it('properly redacts API_KEY (object)', () => {
+      const redacted = censorLog({ apiKey: 'mock-api-key' }, CensorList.getAll())
+      expect(redacted).toEqual({ apiKey: '[API_KEY REDACTED]' })
+    })
+    it('properly redacts API_KEY (multiple nested values)', () => {
+      const redacted = censorLog(
+        { apiKey: 'mock-api-key', config: { headers: { auth: 'mock-api-key' } } },
+        CensorList.getAll(),
+      )
+      expect(redacted).toEqual({
+        apiKey: '[API_KEY REDACTED]',
+        config: { headers: { auth: '[API_KEY REDACTED]' } },
+      })
+    })
```

Comment:
Does it also censor if the value has additional text in addition to the secret? If so, can we add a test for it?

E.g. `Bearer mock-api-key` -> `Bearer [API_KEY REDACTED]`

Reply:
It does so good call! Adding that in.